src/store/fileUploadStore.js

```diff
@@ -282,31 +270,38 @@ const actions = {
 	 * @param {Function} context.dispatch the contexts dispatch function.
 	 * @param {object} context.getters the contexts getters object.
 	 * @param {object} context.state the contexts state object.
-	 * @param {string} uploadId The unique uploadId
+	 * @param {object} data the wrapping object
+	 * @param {string} data.uploadId The unique uploadId
+	 * @param {string} [data.caption] The text caption to the media
 	 */
-	async uploadFiles({ commit, dispatch, state, getters }, uploadId) {
+	async uploadFiles({ commit, dispatch, state, getters }, { uploadId, caption }) {
 		if (state.currentUploadId === uploadId) {
 			commit('setCurrentUploadId', undefined)
 		}
 
 		EventBus.$emit('upload-start')
 
-		// Tag the previously indexed files and add the temporary messages to the
-		// messages list
-		for (const index in state.uploads[uploadId].files) {
+		// Tag previously indexed files and add temporary messages to the MessagesList
+		// If caption is provided, attach to the last temporary message
+		const lastIndex = getters.getUploadsArray(uploadId).at(-1).at(0)
+		for (const [index, uploadedFile] of getters.getUploadsArray(uploadId)) {
 			// mark all files as uploading
 			commit('markFileAsUploading', { uploadId, index })
 			// Store the previously created temporary message
-			const temporaryMessage = state.uploads[uploadId].files[index].temporaryMessage
+			const temporaryMessage = {
+				...uploadedFile.temporaryMessage,
+				message: index === lastIndex ? caption : '{file}',
```

Comment:
Maybe it's better to store the rendered message format here and use it in `Message` . After all, the message contains a file and usually there is {file} to indicate its presence in `Message.message`

Reply:
We got either 'caption' or '{file}' in the server response, so temp message should follow the same logic here, IMO